packages/edit-site/src/components/global-styles/screen-revisions.js

```diff
@@ -0,0 +1,163 @@
+import { isGlobalStyleConfigEqual } from './utils';
+
+function RevisionsSelect( { userRevisions, currentRevisionId, onChange } ) {
+	const userRevisionsOptions = useMemo( () => {
+		return ( userRevisions ?? [] ).map( ( revision ) => {
+			return {
+				value: revision.id,
+				label: decodeEntities( revision.title.rendered ),
+			};
+		} );
+	}, [ userRevisions ] );
+	const setCurrentRevisionId = ( value ) => {
+		const revisionId = Number( value );
+		onChange(
+			userRevisions.find( ( revision ) => revision.id === revisionId )
+		);
+	};
+	return (
+		<SelectControl
+			className="edit-site-global-styles-screen-revisions__selector"
+			label={ __( 'Styles revisions' ) }
+			options={ userRevisionsOptions }
+			onChange={ setCurrentRevisionId }
+			value={ currentRevisionId }
+		/>
+	);
+}
+
+function RevisionsButtons( { userRevisions, currentRevisionId, onChange } ) {
+	return (
+		<ol className="edit-site-global-styles-screen-revisions__revisions-list">
+			{ userRevisions.map( ( revision ) => {
+				const isActive = revision?.id === currentRevisionId;
+				return (
+					<li key={ `user-styles-revision-${ revision.id }` }>
+						<Button
+							className={ classnames(
+								'edit-site-global-styles-screen-revisions__revision-item',
+								{
+									'is-current': isActive,
+								}
+							) }
+							variant={ isActive ? 'tertiary' : 'secondary' }
+							disabled={ isActive }
+							onClick={ () => {
+								onChange( revision );
+							} }
+						>
+							<span className="edit-site-global-styles-screen-revisions__title">
+								{ revision.title.rendered }
+							</span>
+						</Button>
+					</li>
+				);
+			} ) }
+		</ol>
+	);
+}
+
+function ScreenRevisions() {
+	const { user: userConfig, setUserConfig } =
+		useContext( GlobalStylesContext );
+	const { userRevisions } = useSelect(
+		( select ) => ( {
+			userRevisions:
+				select(
+					coreStore
+				).__experimentalGetCurrentThemeGlobalStylesRevisions() || [],
+		} ),
+		[]
+	);
+	const [ currentRevisionId, setCurrentRevisionId ] = useState();
+	const hasRevisions = userRevisions.length > 0;
+
+	useEffect( () => {
+		if ( ! hasRevisions ) {
+			return;
+		}
+		let currentRevision = userRevisions[ 0 ];
+		for ( let i = 0; i < userRevisions.length; i++ ) {
+			if ( isGlobalStyleConfigEqual( userConfig, userRevisions[ i ] ) ) {
+				currentRevision = userRevisions[ i ];
+				break;
+			}
+		}
+		setCurrentRevisionId( currentRevision?.id );
+	}, [ userRevisions, hasRevisions ] );
+
+	const restoreRevision = useCallback(
+		( revision ) => {
+			const newUserConfig = ! userConfig
+				? {}
+				: JSON.parse( JSON.stringify( userConfig ) );
+			set( newUserConfig, [ 'styles' ], revision?.styles );
+			set( newUserConfig, [ 'settings' ], revision?.settings );
+			setUserConfig( () => newUserConfig );
+			setCurrentRevisionId( revision?.id );
+		},
+		[ userConfig ]
+	);
+
+	const RevisionsComponent =
+		userRevisions.length >= 10 ? RevisionsSelect : RevisionsButtons;
```

Comment:
We're not actually showing more than the 10 latest revisions, is that due to API response limits?

Reply:
Oh, I wanted to bump up the max number in the response to `100` but it turns out that I [forgot](https://github.com/WordPress/gutenberg/pull/46667/files#diff-265a1cbfa316459a657c81af12ad3f459bb2808f71758d11c163332bcb025742R129) 🤦